fatal error: jni.h: No such file or directory with Bazel@HEAD

https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3519#018c569e-3795-4226-ab3e-197736d4484f

Platform : Windows, Ubuntu, MacOS

Logs:
```
app/src/main/cpp/native-lib.cpp:1:10: fatal error: jni.h: No such file or directory
 #include <jni.h>
          ^~~~~~~
compilation terminated.
(02:09:02) ERROR: /var/lib/buildkite-agent/builds/bk-docker-l9gw/bazel-org-repo-root/examples/android/ndk/app/src/main/BUILD.bazel:17:11: Compiling app/src/main/cpp/native-lib.cpp failed: (Exit 1): gcc failed: error executing CppCompile command (from target //app/src/main:jni_lib)
```
Steps:
```
git clone -v https://github.com/bazelbuild/examples.git
git reset e2746bf0e3f987cc556e5188c18fec33edc1b61a --hard
export USE_BAZEL_VERSION=591d1257c51c2f997e067964878e3a9c7ee1f89c
cd /android/ndk
bazel build --noincompatible_enable_android_toolchain_resolution --fat_apk_cpu=arm64-v8a,x86 --android_crosstool_top=@androidndk//:toolchain --enable_bzlmod  //app/src/main:app
```
CC Greenteam @comius

--- a/.bazelci/android.yml
+++ b/.bazelci/android.yml
@@ -5,7 +5,6 @@ common:
     bazel: last_green
     working_directory: ../android/firebase-cloud-messaging
     build_flags:
-      - "--incompatible_enable_android_toolchain_resolution"
       - "--android_platforms=//:arm64-v8a"
     build_targets:
       - "//app:app"
@@ -14,11 +13,7 @@ common:
     bazel: 6.3.2
     working_directory: ../android/ndk
     build_flags:
-      # TODO(https://github.com/bazelbuild/rules_android/issues/77):
-      #   Enable Android Platforms.
-      - "--noincompatible_enable_android_toolchain_resolution"
-      - "--fat_apk_cpu=arm64-v8a,x86"
-      - "--android_crosstool_top=@androidndk//:toolchain"
+      - "--android_platforms=//:arm64-v8a,//:x86"
       - "--enable_bzlmod"
     build_targets:
       - "//app/src/main:app"
@@ -28,7 +23,7 @@ common:
     bazel: 6.3.2
     working_directory: ../android/jetpack-compose
     build_flags:
-      - "--noincompatible_enable_android_toolchain_resolution"
+      - "--android_platforms=//:arm64-v8a"
       - "--remote_download_outputs=all"
       - "--enable_bzlmod"
     build_targets:
@@ -39,7 +34,7 @@ common:
     bazel: 6.3.2
     working_directory: ../android/robolectric-testing
     test_flags:
-      - "--noincompatible_enable_android_toolchain_resolution"
+      - "--android_platforms=//:arm64-v8a"
       - "--enable_bzlmod"
     test_targets:
       - "//app:test"

--- a/android/jetpack-compose/BUILD.bazel
+++ b/android/jetpack-compose/BUILD.bazel
@@ -7,3 +7,11 @@ kt_compiler_plugin(
     visibility = ["//visibility:public"],
     deps = ["@maven//:androidx_compose_compiler_compiler"],
 )
+
+platform(
+    name = "arm64-v8a",
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:android",
+    ],
+)

--- a/android/ndk/BUILD.bazel
+++ b/android/ndk/BUILD.bazel
@@ -1,0 +1,15 @@
+platform(
+    name = "arm64-v8a",
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:android",
+    ],
+)
+
+platform(
+    name = "x86",
+    constraint_values = [
+        "@platforms//cpu:x86_32",
+        "@platforms//os:android",
+    ],
+)

--- a/android/robolectric-testing/BUILD.bazel
+++ b/android/robolectric-testing/BUILD.bazel
@@ -1,0 +1,7 @@
+platform(
+    name = "arm64-v8a",
+    constraint_values = [
+        "@platforms//cpu:arm64",
+        "@platforms//os:android",
+    ],
+)

